tests/integration/conftest.py

```diff
@@ -55,3 +55,25 @@ async def override_get_db():
     # For debugging change rollback to commit.
     transaction.rollback()
     connection.close()
+
+
+@pytest.fixture
+def create_ensemble(client):
+    def func(experiment_id, parameters=None):
+        if parameters is None:
+            parameters = []
+        resp = client.post_check(
+            f"/experiments/{experiment_id}/ensembles", json={"parameters": parameters}
+        )
+        return resp.json()["id"]
+
+    return func
```

Comment:
Seems like in all the tests, we are dealing with only a single experiment. Let's make this fixture easier to work with by using the `request` fixture and the current running test name with `request.node.name`:

```suggestion
@pytest.fixture
def create_ensemble(client, request):
    def func(experiment_id=None, parameters=None):
        if experiment_id is None:
            experiment_id = client.post_check("/experiments", json={"name": request.node.name}).json()["id"]
        if parameters is None:
            parameters = []
        resp = client.post_check(
            f"/experiments/{experiment_id}/ensembles", json={"parameters": parameters}
        )
        return resp.json()["id"]

    return func
```

Then you only need to use `create_ensemble`, which will create an experiment for you with the name of the test.

Reply:
This does not work if we want to have multiple ensembles in an experiment

Reply:
Reworked it :)